### PR TITLE
[HOLD] Actually use clang for the build.

### PR DESCRIPTION
**Pull request by tylerkaraszewski**
### Details
[HOLD] https://github.com/Expensify/Bedrock/pull/1837 https://github.com/Expensify/Bedrock/pull/1838 And whatever other switchover PRs we need to move to clang everywhere.

~~SSL test and CHainedHTTP`BadCommandTest` test currently fails (in GH Actions but not dev).~~ Tests are fixed.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/411196

### Tests

_________
**Internal Testing Reminder:** when changing bedrock, please compile auth against your new changes

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,10 +3,10 @@
 # to be set, but for the time being we need to override the defaults so that our existing dev environment works. This
 # can be removed when that is resolved.
 ifeq ($(CC),cc)
-CC = gcc-13
+CC = ccache /usr/bin/clang-18
```
Comment (bondydaa) [inline Makefile:6]: testing this locally and commenting out these out in my .bashrc:

```
export CC="ccache /usr/bin/clang-18"
export CXX="ccache /usr/bin/clang++-18"
```

still led to `make` using `g++13`
```
vagrant@expensidev2004:/vagrant/Bedrock$ make -j8
g++-13 -g -std=c++20 -fPIC -DSQLITE_ENABLE_NORMALIZE -O0 -Wall -Werror -Wformat-security -Wno-unqualified-std-cast-call -Wno-error=deprecated-declarations -I/vagrant/Bedrock -I/vagrant/Bedrock/mbedtls/include -MMD -MF .build/test/clustertest/testplugin/ExternPointer.d -MT .build/test/clustertest/testplugin/ExternPointer.o -o .build/test/clustertest/testplugin/ExternPointer.o -c test/clustertest/testplugin/ExternPointer.cpp
g++-13 -g -std=c++20 -fPIC -DSQLITE_ENABLE_NORMALIZE -O0 -Wall -Werror -Wformat-security -Wno-unqualified-std-cast-call -Wno-error=deprecated-declarations -I/vagrant/Bedrock -I/vagrant/Bedrock/mbedtls/include -MMD -MF .build/test/clustertest/testplugin/TestPlugin.d -MT .build/test/clustertest/testplugin/TestPlugin.o -o .build/test/clustertest/testplugin/TestPlugin.o -c test/clustertest/testplugin/TestPlugin.cpp
```

I believe b/c
```
vagrant@expensidev2004:/vagrant/Bedrock$ echo $CXX
g++-13
```

but I'm not exactly sure where that is getting set, maybe something in expensidev?

At least 👍 this build properly with clang on GH Actions 
image

**Review by bondydaa — APPROVED** [review]
approving, i assume we need to add something expensidev (or an action required email) to get everyone to add it locally so we end up using clang vs g++?

**Comment by tylerkaraszewski** [comment]
> approving, i assume we need to add something expensidev (or an action required email) to get everyone to add it locally so we end up using clang vs g++?

Yeah, we need similar cahnges in auth, jenkins, etc.

File: Makefile (hunk 1 above)
Comment (tylerkaraszewski) [reply Makefile:6]: Did you close the shell and reopen a new one after commenting those lines out? They will still be set in your existing shell and any new shells opened from it until you exit.

**Pull request closed without merging**